### ISSUE TITLE
NuGet Package

### ISSUE THREAD

**Issue by yozzozo** [OneSignal/OneSignal-Xamarin-SDK#9]
Hi,

Can we please get the OneSignal Xamarin SDK as a NuGet package?

It will be far easier to maintain / update as a NuGet package rather than something you have to manually check for updates, download, and re-integrate every time.

Thanks!

**Comment by SeeD-Seifer** [comment]
Vote for this too

**Comment by jzeferino** [comment]
Vote for it!

**Comment by robtaireeo** [comment]
+1

**Comment by Aehrraid** [comment]
+1

**Comment by rudyryk** [comment]
+1

**Comment by TNPGames** [comment]
I need it too... Because we are using Xamarin + VS and there is to gradle there...

**Comment by stopiccot** [comment]
Yeah, nuget package would be nice to have, current setup involving manually adding projects and files is horrible.

**Comment by deivydas** [comment]
+1

**Comment by Necromunger** [comment]
+1

**Comment by Aehrraid** [comment]
+1

**Comment by ndastur** [comment]
+1

**Comment by ndastur** [comment]
And in the meantime please remove the out of date NuGet package as it just confuses things. Ta

**Comment by escamoteur** [comment]
+1

**Comment by escamoteur** [comment]
you should follow the Xamarin.Plugin strcuture and use the Bait And Switch approach. you can put more than one dll in one nuget package

**Comment by yozzozo** [comment]
Yeah, +1 on that suggestion.  There are plenty of nuget packages that
follow the PCL + bait and switch approach.  Analyze any of the
MvvmCross.Plugins.* packages to see examples.

**Comment by escamoteur** [comment]
There is even a VS Solution template from James MonteMagno for Plugins following this approach.

**Comment by ndastur** [comment]
@jkasten2 to be fair I think we would all prefer 3 seperate projects rather than none at all. The manual integration is killing me!

**Comment by martijn00** [comment]
I've refactored the SDK using Switch and bait, and made a nuspec file ready so the only thing you need to do is publish it on Nuget with the company Nuget api key.

**Comment by ndastur** [comment]
@martijn00 that's great news

**Comment by jkasten2** [comment]
@martijn00 Awesome! Thanks for doing this. Going to test and review it this week.

**Comment by ndastur** [comment]
@martijn00 @jkasten2 hi guys. Any news on this?
Thanks

**Comment by jkasten2** [comment]
@ndastur I have merged and closed the pull request but there is some more work that needs to be done before we release it. There are a bit more details on the pull request. If all goes well I hope to have it released on in the next few days.

**Comment by jkasten2** [comment]
This SDK is now available on Nuget.
https://www.nuget.org/packages/Com.OneSignal/

Setup instructions have been updated on using it.
https://documentation.onesignal.com/docs/xamarin-sdk-setup

**Comment by ndastur** [comment]
Fantastic news. I'll give a try

**Comment by ndastur** [comment]
@martijn00 It was great the NUGet came out but it hasn't been updated in a while. Is there a roadmap?

**Comment by EmilAlipiev** [comment]
@martijn00 thanks for the nuget package. I have been using OneSignal more than a year and i attached source code year ago now I see that nuget requires Playservices.Gcm. why is it required? my old source is still working fine without it. could you please clarify this? thank you
